ui/components/multichain/create-named-snap-account/create-named-snap-account.test.tsx

```diff
@@ -0,0 +1,181 @@
+const mockAccount = createMockInternalAccount({
+  address: mockAddress,
+  name: 'New account',
+});
+const mockSnapSuggestedAccountName = 'Suggested Account Name';
+
+jest.mock('../../../store/actions', () => ({
+  ...jest.requireActual('../../../store/actions'),
+  getNextAvailableAccountName: jest.fn().mockResolvedValue('Snap Account 2'),
+}));
+
+const mockSnapAccount1 = {
+  address: '0xb552685e3d2790efd64a175b00d51f02cdafee5d',
+  id: 'c3deeb99-ba0d-4a4e-a0aa-033fc1f79ae3',
+  metadata: {
+    name: 'Snap Account 1',
+    keyring: {
+      type: 'Snap Keyring',
+    },
+    snap: {
+      id: 'snap-id',
+      name: 'snap-name',
+    },
+  },
+  options: {},
+  methods: ETH_EOA_METHODS,
+  type: 'eip155:eoa',
+};
+const mockSnapAccount2 = {
+  address: '0x3c4d5e6f78901234567890abcdef123456789abc',
+  id: 'f6gccd97-ba4d-4m7e-q3ahj-033fc1f79ae4',
+  metadata: {
+    name: 'Snap Account 2',
+    keyring: {
+      type: 'Snap Keyring',
+    },
+    snap: {
+      id: 'snap-id',
+      name: 'snap-name',
+    },
+  },
+  options: {},
+  methods: ETH_EOA_METHODS,
+  type: 'eip155:eoa',
+};
+
+const render = (
+  props: CreateNamedSnapAccountProps = {
+    onActionComplete: jest.fn().mockResolvedValue({ success: true }),
+    address: mockAccount.address,
+    snapSuggestedAccountName: mockSnapSuggestedAccountName,
+  },
+) => {
+  const store = configureStore({
+    ...mockState,
+    metamask: {
+      ...mockState.metamask,
+      completedOnboarding: true,
+      internalAccounts: {
+        ...mockState.metamask.internalAccounts,
+        accounts: {
+          ...mockState.metamask.internalAccounts.accounts,
+          [mockSnapAccount1.id]: mockSnapAccount1,
+          [mockSnapAccount2.id]: mockSnapAccount2,
+        },
+        options: {},
+        methods: ETH_EOA_METHODS,
+        type: 'eip155:eoa',
+      },
+    },
+  });
+  return renderWithProvider(<CreateNamedSnapAccount {...props} />, store);
+};
+
+describe('CreateNamedSnapAccount', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('displays account name input and suggested name', async () => {
+    const { getByPlaceholderText } = render();
+
+    await waitFor(() =>
+      expect(
+        getByPlaceholderText(mockSnapSuggestedAccountName),
+      ).toBeInTheDocument(),
+    );
+  });
+
+  it('renames account and fires onActionComplete with true when clicking "Add account"', async () => {
+    const onActionComplete = jest.fn();
+    const { getByText, getByPlaceholderText } = render({
+      onActionComplete,
+      address: mockAccount.address,
+      snapSuggestedAccountName: mockSnapSuggestedAccountName,
+    });
+
+    const input = await waitFor(() =>
+      getByPlaceholderText(mockSnapSuggestedAccountName),
+    );
+    const newAccountName = 'New Account Name';
+
+    fireEvent.change(input, {
+      target: { value: newAccountName },
+    });
+    fireEvent.click(getByText('Add account'));
```

Comment:
Nit: We could use `messages.json` like for some other tests in this PR, so that we would use `messages.addAccount` instead, WDYT?

Reply:
This is also true for `Cancel` and other `Add Account` in this file

Reply:
[e117716](https://github.com/MetaMask/metamask-extension/pull/25191/commits/e1177169b62719208f1fc0d3ae3b93f959411d83) 👍